daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/GenReader.scala

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package archive
+
+import com.google.protobuf.{ByteString, CodedInputStream}
+
+import scala.util.Using
+
+final class GenReader[X] private[archive] (val fromCodedInputStream: CodedInputStream => X) {
+
+  // close the stream unconditionally
```

Comment:
is that deliberate? as mentioned before I generally find code easier to reason about if the code responsible for opening a stream is responsible for closing it as well.

Reply:
Following the comment on the other PR, I removed the `close`, but forgot the comment.